Reproducible failure in TestGrouping.testRandom

### Description

  >     java.lang.AssertionError: expected.totalGroupedHitCount != actual.totalGroupedHitCount expected:<23> but was:<21>
   >         at __randomizedtesting.SeedInfo.seed([4BD207D49F54EEDC:399E22DB2E3458AF]:0)
   >         at junit@4.13.1/org.junit.Assert.fail(Assert.java:89)
   >         at junit@4.13.1/org.junit.Assert.failNotEquals(Assert.java:835)
   >         at junit@4.13.1/org.junit.Assert.assertEquals(Assert.java:647)
   >         at org.apache.lucene.search.grouping.TestGrouping.assertEquals(TestGrouping.java:1516)
   >         at org.apache.lucene.search.grouping.TestGrouping.testRandom(TestGrouping.java:1313)
   >         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
   >         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
   >         at junit@4.13.1/org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
   >         at org.apache.lucene.test_framework@10.0.0-SNAPSHOT/org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
   >         at junit@4.13.1/org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
   >         at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
   >         at java.base/java.lang.Thread.run(Thread.java:1583)

### Gradle command to reproduce

./gradlew test --tests TestGrouping.testRandom -Dtests.seed=4BD207D49F54EEDC -Dtests.nightly=true -Dtests.locale=es-CR -Dtests.timezone=Africa/Malabo -Dtests.asserts=true -Dtests.file.encoding=UTF-8

I hit this when testing the upcoming 9.9.2 release.

```
org.apache.lucene.search.grouping.TestGrouping > testRandom FAILED
  java.lang.AssertionError: expected.totalGroupedHitCount != actual.totalGroupedHitCount expected:<77> but was:<79>
    at __randomizedtesting.SeedInfo.seed([CDCE9F629E2B356F:BF82BA6D2F4B831C]:0)
    at junit@4.13.1/org.junit.Assert.fail(Assert.java:89)
    at junit@4.13.1/org.junit.Assert.failNotEquals(Assert.java:835)
    at junit@4.13.1/org.junit.Assert.assertEquals(Assert.java:647)
    at org.apache.lucene.search.grouping.TestGrouping.assertEquals(TestGrouping.java:1516)
    at org.apache.lucene.search.grouping.TestGrouping.testRandom(TestGrouping.java:1313)
    ...
```

```
./gradlew :lucene:grouping:test --tests "org.apache.lucene.search.grouping.TestGrouping.testRandom" -Ptests.seed=CDCE9F629E2B356F -Ptests.nightly=true
```

Ok, it fails on _main_ when run with the see from @easyice 

```
./gradlew test --tests TestGrouping.testRandom -Dtests.seed=4BD207D49F54EEDC -Dtests.nightly=true -Dtests.locale=es-CR -Dtests.timezone=Africa/Malabo -Dtests.asserts=true -Dtests.file.encoding=UTF-8
 java.lang.AssertionError: expected.totalGroupedHitCount != actual.totalGroupedHitCount expected:<23> but was:<21>
```

--- a/lucene/grouping/src/test/org/apache/lucene/search/grouping/TestGrouping.java
+++ b/lucene/grouping/src/test/org/apache/lucene/search/grouping/TestGrouping.java
@@ -611,7 +611,11 @@ public class TestGrouping extends LuceneTestCase {
     }
 
     RandomIndexWriter w =
-        new RandomIndexWriter(random(), dir, newIndexWriterConfig(new MockAnalyzer(random())));
+        new RandomIndexWriter(
+            random(),
+            dir,
+            newIndexWriterConfig(new MockAnalyzer(random()))
+                .setMergePolicy(newMergePolicy(random(), false)));
 
     final List<List<Document>> updateDocs = new ArrayList<>();
 
@@ -731,7 +735,11 @@ public class TestGrouping extends LuceneTestCase {
 
       Directory dir = newDirectory();
       RandomIndexWriter w =
-          new RandomIndexWriter(random(), dir, newIndexWriterConfig(new MockAnalyzer(random())));
+          new RandomIndexWriter(
+              random(),
+              dir,
+              newIndexWriterConfig(new MockAnalyzer(random()))
+                  .setMergePolicy(newMergePolicy(random(), false)));
       Document doc = new Document();
       Document docNoGroup = new Document();
       Field idvGroupField = new SortedDocValuesField("group", new BytesRef());

